src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,250 @@ class C1<T>
+
+            var param1Original = ptrOriginal.Signature.Parameters[0];
+            var param2Original = ptrOriginal.Signature.Parameters[1];
+            var param1Retargeted = ptrRetargeted.Signature.Parameters[0];
+            var param2Retargeted = ptrRetargeted.Signature.Parameters[1];
+
+            assert(param1Consistent.typeConsistent,
+                   param1Original.Type.ContainingAssembly,
+                   param1Retargeted.Type.ContainingAssembly);
+            assert(param1Consistent.refModConsistent,
+                   getModifierAssemblySymbol(param1Original.RefCustomModifiers),
+                   getModifierAssemblySymbol(param1Retargeted.RefCustomModifiers));
+            assert(param1Consistent.typeModConsistent,
+                   getModifierAssemblySymbol(param1Original.TypeWithAnnotations.CustomModifiers),
+                   getModifierAssemblySymbol(param1Retargeted.TypeWithAnnotations.CustomModifiers));
+
+            assert(param2Consistent.typeConsistent,
+                   param2Original.Type.ContainingAssembly,
+                   param2Retargeted.Type.ContainingAssembly);
+            assert(param2Consistent.refModConsistent,
+                   getModifierAssemblySymbol(param2Original.RefCustomModifiers),
+                   getModifierAssemblySymbol(param2Retargeted.RefCustomModifiers));
+            assert(param2Consistent.typeModConsistent,
+                   getModifierAssemblySymbol(param2Original.TypeWithAnnotations.CustomModifiers),
+                   getModifierAssemblySymbol(param2Retargeted.TypeWithAnnotations.CustomModifiers));
+
+            static MethodSymbol getMethodSymbol(CSharpCompilation compilation)
+            {
+                var c = compilation.GetTypeByMetadataName("Source");
+                return c.GetMethod("M");
+            }
+
+            static AssemblySymbol getModifierAssemblySymbol(ImmutableArray<CustomModifier> modifiers)
+                => ((CSharpCustomModifier)modifiers.Single()).ModifierSymbol.ContainingAssembly;
+
+            void assert(bool consistent, AssemblySymbol original, AssemblySymbol retargeted)
+            {
+                if (consistent)
+                {
+                    Assert.Same(consistentAssembly, original);
+                    Assert.Same(consistentAssembly, retargeted);
+                }
+                else
+                {
+                    Assert.Same(retargetedAssembly1, original);
+                    Assert.Same(retargetedAssembly2, retargeted);
+                }
+            }
+
+            static (AssemblySymbol retargetedAssembly1, AssemblySymbol retargetedAssembly2, AssemblySymbol consistentAssembly, CSharpCompilation originalComp, CSharpCompilation retargetedComp)
+                getFunctionPointerRetargetingDefinitions(string mIlSignature, string mOverriddenSignature)
+            {
+                var retargetedSource = @"
+using System.Reflection;
+[assembly:AssemblyVersionAttribute(""{0}"")]]
```

Comment:
This feels more complicated than it could be. CreateCompilation takes an AssemblyIdentity instance (at least there is an overload that does), this how usually we specify version in tests rather than changing/parameterizing the source.